docs/configuring-a-resource.md

```diff
@@ -0,0 +1,410 @@
+type of the resource. Since, `AccessKey` and `User` resources are under the same
+go package, we don't need to provide the package path. However, this is not
+always the case and referenced resources might be in different package. In that
+case, we would need to provide the full path. Referencing to a [kms key] from
+`aws_ebs_volume` resource is a good example here:
+
+```go
+func Configure(p *config.Provider) {
+	p.AddResourceConfigurator("aws_ebs_volume", func(r *config.Resource) {
+		r.References["kms_key_id"] = config.Reference{
+			Type: "github.com/crossplane-contrib/provider-tf-aws/apis/kms/v1alpha1.Key",
+		}
+	})
+}
+```
+
+### Additional Sensitive Fields and Custom Connection Details
+
+Crossplane stores sensitive information of a managed resource in a Kubernetes
+secret, together with some additional fields that would help consumption of the
+resource, a.k.a. [connection details].
+
+In Terrajet, we already [handle sensitive fields] that are marked as sensitive
+in Terraform schema and no further action required for them. However, we still
+have some custom configuration API that would allow marking additional fields as
+sensitive (e.g. just if we encounter a field that is not marked properly) and
+also to [add additional keys with custom values] no matter they are sensitive or
+not.
+
+As an example, let's use `aws_iam_access_key`. Currently, Terrajet stores all
+sensitive fields in Terraform schema as prefixed with `attribute.`, so without
+any `AdditionalConnectionDetailsFn`, connection resource will have
+`attribute.id` and `attribute.secret` corresponding to [id] and [secret] fields
+respectively. To see them with more common keys, i.e. `aws_access_key_id` and 
+`aws_secret_access_key`, we would need to make the following configuration:
+
+```go
+func Configure(p *config.Provider) {
+	p.AddResourceConfigurator("aws_iam_access_key", func(r *config.Resource) {
+		r.Sensitive = config.Sensitive{
+			AdditionalConnectionDetailsFn: func(attr map[string]interface{}) (map[string][]byte, error) {
+				conn := map[string][]byte{}
+				if a, ok := attr["id"].(string); ok {
+					conn["aws_access_key_id"] = []byte(a)
+				}
+				if a, ok := attr["secret"].(string); ok {
+					conn["aws_secret_access_key"] = []byte(a)
+				}
+				return conn, nil
+			},
+		}
+	})
+}
+```
+
+This will produce a connection details secret as follows:
+
+```yaml
+apiVersion: v1
+data:
+  attribute.id: QUtJQVk0QUZUVFNFNDI2TlhKS0I=
+  attribute.secret: ABCxyzRedacted==
+  attribute.ses_smtp_password_v4: QQ00REDACTED==
+  aws_access_key_id: QUtJQVk0QUZUVFNFNDI2TlhKS0I=
+  aws_secret_access_key: ABCxyzRedacted==
+kind: Secret
+```
+
+### Late Initialization Behavior
+Terrajet runtime automatically performs late-initialization during
+an [`external.Observe`] call with means of runtime reflection.
+State of the world observed by Terraform CLI is used to initialize
+any `nil`-valued pointer parameters in the managed resource's `spec`.
```

Comment:
I think this is too technical of a description late-init. Maybe saying something along the lines of `it will fill the blanks in spec by defaults from the cloud provider` would be easier to understand what it's for.

Reply:
@ulucinar what do you think? 
In any case, I would prefer to postpone rewdoring this to #153